### PR TITLE
Add cloud-init build feature

### PR DESCRIPTION
**Pull request by rhatdan**
Improve help and README to describe new functionality.

### CODE REVIEW

**Comment by rhatdan** [comment]
@n1hility @cgwalters PTAL

File: training/Makefile [hunk 1]
```diff
@@ -1,15 +1,71 @@
...
+	make -C nvidia-bootc/ dtk bootc
+vllm:
+	make -C vllm/ image
+
+#
+# Make Bootc container images preinstalled with cloud-init
+#
+.PHONY:
+cloud-amd:
+	make VENDOR=amd -C cloud
```
Comment (cgwalters) [inline training/Makefile:57]: Is there a missing `git add`?  I don't see this directory in the PR

File: training/README.md [hunk 2]
```diff
@@ -57,6 +65,15 @@ Of course, the other Makefile variables are still available, so the following is
 make nvidia REGISTRY=myregistry.com REGISTRY_ORG=ai-training IMAGE_NAME=nvidia IMAGE_TAG=v1 FROM=registry.redhat.io/rhel9/rhel-bootc:9.4
 ```
 
+# How to build Cloud ready images
+
+Bootc container images can be installed on physical machines, virtual machines and in the cloud.  Often it is useful to add the cloud-init package when running the operarting systems in the cloud.
```
Comment (cgwalters) [inline training/README.md:70]: I think we can shortly redirect to some upstream docs on this btw.

But the larger problem is that cloud-init isn't currently working for the growfs case so we should do https://github.com/containers/ai-lab-recipes/pull/410 first.

File: training/README.md [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
-| intel       | Create bootable container for Intel Habanalabs platform |
+| Target          | Description                                                         |
+|-----------------|---------------------------------------------------------------------|
+| instruct-amd    | Create instruct lab image for bootable container for AMD platform   |
+| instruct-intel  | Create instruct lab image for bootable container for Intel platform |
+| instruct-nvidia | Create instruct lab image for bootable container for Nvidia platform|
+| amd             | Create bootable container for AMD platform                          |
+| intel           | Create bootable container for Intel Habanalabs platform             |
+| nvidia          | Create bootable container for NVidia platform                       |
+| cloud-amd       | Add cloud-init to bootable container for AMD platform               |
```
Comment (cgwalters) [inline training/README.md:16]: It feels like it'd be nicer to have this as flag, something like

```
make amd INCLUDES=cloud-init
```

Would generalize more obviously to including other snippets in the future.

File: training/Makefile (hunk 1 above)
Comment (rhatdan) [reply training/Makefile:57]: Yup added.

File: training/README.md (hunk 3 above)
Comment (rhatdan) [reply training/README.md:16]: Sounds good lets merge and then I can work on this. Would involve more complicated Makefile gymnastics.

**Pull request merged**